File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    
    q1 = deque(queue1)
    q2 = deque(queue2)
    
    sum1 = sum(q1)
    sum2 = sum(q2)
    
    if (sum1+sum2)%2 == 1:
        return -1
    
    target = (sum1+sum2)//2
    max_len = 2*(len(q1)+len(q2))
    
    while answer < max_len:
        if sum1 == target:
            return answer
        
        if sum1 > target:
            out = q1.popleft()
            sum1 -= out
            q2.append(out)
            sum2 += out
        else:
            out = q2.popleft()
            sum2 -= out
            q1.append(out)
            sum1 += out
        
        answer += 1
    
    
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        long sum1 = 0;
        long sum2 = 0;
        
        Deque<Integer> q1 = new ArrayDeque<>();
        Deque<Integer> q2 = new ArrayDeque<>();
        
        for(int a: queue1){
            sum1 += a;
            q1.addLast(a);
        }
        
        for(int b: queue2){
            sum2 += b;
            q2.addLast(b);
        }
        
        if((sum1+sum2)%2 == 1) return -1; // 합이 홀수면 불가능

        long target = (sum1+sum2)/2;
        int maxLen = 2*(q1.size() + q2.size());
        
        while(answer < maxLen){
            if(sum1 == target) return answer;
            
            if(sum1 > target){
                int out = q1.pollFirst();
                sum1 -= out;
                q2.addLast(out);
                sum2 += out;
            } else {
                int out = q2.pollFirst();
                sum2 -= out;
                q1.addLast(out);
                sum1 += out;
            }
            answer++;
        }
        return -1;
    }
}